[package] libmysqlclient/8.0.25: fails to build on latest macOS 12.0.1

<!-- 
libmysqlclient/8.0.25 fails to build on latest macOS 12.0.1 with clang 13
-->

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libmysqlclient/8.0.25**
  * Operating System+version: macOS 12.0.1
  * Compiler+version: **clang 13**
  * Conan version: **conan 1.42.0**
  * Python version: **Python 3.9.7**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13.0
os=Macos
os_build=Macos
[options]
[build_requires]
[env]
```


### Steps to reproduce (Include if Applicable)



### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
CMake Error at cmake/package_name.cmake:98 (MESSAGE):
  Could not run sw_vers
Call Stack (most recent call first):
  cmake/package_name.cmake:152 (GET_PACKAGE_FILE_NAME)
  CMakeListsOriginal.txt:542 (INCLUDE)
  CMakeLists.txt:8 (include)
```

</details>

I think the reason is in the file of: 
```
.conan/data/libmysqlclient//8.0.25/_/_/source/source_subfolder/cmake/package_name.cmake
```

The code of checking the version:

```
      STRING(REGEX MATCH
        "ProductVersion:[\n\t ]*([0-9]+)\\.([0-9]+)" UNUSED ${SW_VERS_PRODUCTVERSION})
      IF(NOT CMAKE_MATCH_1 OR NOT CMAKE_MATCH_2)
        MESSAGE(FATAL_ERROR "Could not run sw_vers")
```
In this case `CMAKE_MATCH_1` will be `12` while `CMAKE_MATCH_2` will be `0`, so the build always fails

Should this be patched on the conan package level or updated in libmysqlclient repo?

--- a/recipes/libmysqlclient/all/conandata.yml
+++ b/recipes/libmysqlclient/all/conandata.yml
@@ -16,3 +16,5 @@ patches:
   "8.0.25":
     - base_path: "source_subfolder"
       patch_file: "patches/0004-fix-805-cpp17-build.patch"
+    - base_path: "source_subfolder"
+      patch_file: "patches/0005-fix-macos-12.0.x-version-detection.patch"

--- a/recipes/libmysqlclient/all/patches/0005-fix-macos-12.0.x-version-detection.patch
+++ b/recipes/libmysqlclient/all/patches/0005-fix-macos-12.0.x-version-detection.patch
@@ -1,0 +1,13 @@
+diff --git a/cmake/package_name.cmake b/cmake/package_name.cmake
+index 192035dd..dedc59c6 100644
+--- a/cmake/package_name.cmake
++++ b/cmake/package_name.cmake
+@@ -94,7 +94,7 @@ MACRO(GET_PACKAGE_FILE_NAME Var)
+ 
+       STRING(REGEX MATCH
+         "ProductVersion:[\n\t ]*([0-9]+)\\.([0-9]+)" UNUSED ${SW_VERS_PRODUCTVERSION})
+-      IF(NOT CMAKE_MATCH_1 OR NOT CMAKE_MATCH_2)
++      IF(CMAKE_MATCH_1 STREQUAL "" OR CMAKE_MATCH_2 STREQUAL "")
+         MESSAGE(FATAL_ERROR "Could not run sw_vers")
+       ENDIF()
+ 

